provisioner/data.js

```diff
@@ -86,6 +87,107 @@ var WorkerType = base.Entity.configure({
+    canUseSpot: base.Entity.types.JSON, // delete this
+    instanceTypes: base.Entity.types.JSON,
+    regions: base.Entity.types.JSON,
+    // Store the date of last modification for this entity
+    lastModified: base.Entity.types.Date,
+    // Global base UserData object for overwriting
+    userData: base.Entity.types.JSON,
+    // Global base LaunchSpecification object for overwriting
+    launchSpec: base.Entity.types.JSON,
+    // Global base secrets object for overwriting
+    secrets: base.Entity.types.JSON,
+    // Global base scope list for appending, these are the scopes that
+    // temporary TC credentials will be issued against
+    scopes: base.Entity.types.JSON,
+  },
+  migrate: function (item) {
+    console.log('Upgrading a workerType to version 2');
+    console.log(JSON.stringify(item, null, 2));
+    var oldUserData;
+    if (item.launchSpecification.UserData) {
+      oldUserData = JSON.parse(new Buffer(item.launchSpecification.UserData, 'base64').toString());
+    } else {
+      oldUserData = {};
+    }
+    var oldLaunchSpec = item.launchSpecification;
+    delete item.launchSpecification;
+    delete oldLaunchSpec.UserData;
+    item.launchSpec = lodash.clone(oldLaunchSpec);
+    item.secrets = {};
+    item.scopes = [];
+    item.userData = lodash.clone(oldUserData);
+
+    // Now, strip out the UserData and LaunchSpec stuff from existing entries
+    // and put them in the right structure then create empty scopes and secrets
+    item.regions.forEach(r => {
+      // Let's get the old launch spec and user data
+      var oldUserData;
+      if (r.overwrites && r.overwrites.UserData) {
+        oldUserData = JSON.parse(new Buffer(r.overwrites.UserData, 'base64').toString());
+        ['capacity', 'provisionerId', 'workerType'].forEach(key => {
+          if (oldUserData[key]) {
+            delete oldUserData[key];
+          }
+        });
+      } else {
+        oldUserData = {};
+      }
+      var oldOverwrites = r.overwrites;
+      delete r.overwrites;
+
+      // Now set them up
+      r.userData = oldUserData;
+      r.launchSpec = oldOverwrites;
+      r.secrets = {};
+      r.scopes = [];
+
+    });
+
+    item.instanceTypes.forEach(t => {
+      // Let's get the old launch spec and user data
+      var oldUserData;
+      if (t.overwrites && t.overwrites.UserData) {
+        oldUserData = JSON.parse(new Buffer(t.overwrites.UserData, 'base64').toString());
+        ['capacity', 'provisionerId', 'workerType'].forEach(key => {
+          if (oldUserData[key]) {
+            delete oldUserData[key];
+          }
+        });
+      } else {
+        oldUserData = {};
+      }
+      var oldOverwrites = t.overwrites;
+      delete t.overwrites;
+
+      // Now set them up
+      t.userData = oldUserData;
+      t.launchSpec = oldOverwrites;
+      t.secrets = {};
+      t.scopes = [];
+    });
+
+    item.lastModified = new Date();
+    console.log('===>');
+    console.log(JSON.stringify(item, null, 2));
```

Comment:
yep! the intention was to delete this before going into production as one of the cleanups for the final review.